File: Java/PdfViewerAPI_R2/java/com/pdf_tools/pdfviewer/Requests/PdfGetOpenActionDestinationRequest.java
package com.pdf_tools.pdfviewer.Requests;

import com.pdf_tools.pdfviewer.DocumentManagement.IPdfDocument;
import com.pdf_tools.pdfviewer.Model.IPdfControllerCallbackManager;
import com.pdf_tools.pdfviewer.Model.PdfAdvancedDestination;
import com.pdf_tools.pdfviewer.Model.PdfViewerException;

public class PdfGetOpenActionDestinationRequest extends APdfRequest<PdfAdvancedDestination>
{

    public PdfGetOpenActionDestinationRequest()
    {
        this._priority = 88;
    }

    @Override
    public void execute(IPdfDocument document, IPdfControllerCallbackManager controller)
    {
        try
        {
            PdfAdvancedDestination dest = document.getOpenActionDestination();
            completedEvent.triggerEvent(dest, null);
        } catch (PdfViewerException ex)
        {
            completedEvent.triggerEvent(null, ex);
        }
    }

}

File: CS.NET/PdfViewerCSharpAPI_R2/DocumentManagement/Requests/PdfGetOpenActionDestinationRequest.cs
// -----------------------------------------------------------------------
// <copyright file="PdfGetOpenActionDestinationRequest.cs" company="">
// TODO: Update copyright text.
// </copyright>
// -----------------------------------------------------------------------

namespace PdfTools.PdfViewerCSharpAPI.DocumentManagement.Requests
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;
    using PdfTools.PdfViewerCSharpAPI.Utilities;
    using PdfTools.PdfViewerCSharpAPI.Model;

    /// <summary>
    /// TODO: Update summary.
    /// </summary>
    public class PdfGetOpenActionDestinationRequest : APdfRequest<object, PdfDestination>
    {
        public PdfGetOpenActionDestinationRequest()
            : base(null, 88)
        {
        }

        protected override PdfDestination ExecuteNative(IPdfDocument document, object nothing)
        {
            return document.GetOpenActionDestination();
        }


        protected override void triggerControllerCallback(IPdfControllerCallbackManager controller, InOutTuple tuple, PdfViewerException ex)
        {
        }
        protected override void triggerControllerCallback(IPdfControllerCallbackManager controller, PdfViewerException ex)
        {
        }
    }
}
